doc/CMakeLists.txt

```diff
@@ -21,8 +21,17 @@ if( EXISTS ${XELATEX} )
   # add a configure rule to the header file though
   configure_file(${PROJECT_SOURCE_DIR}/title.tex.in ${PROJECT_SOURCE_DIR}/title.tex)
 
-  # Add custom target to build only documentation
-  add_custom_target(docs ALL)
+  # Add custom target to build only documentation, this target will depend on all individual PDFs
+  get_directory_property(hasParent PARENT_DIRECTORY)
+  if(hasParent AND BUILD_DOCS_ONLY_WITH_PACKAGE)
+    # Because we don't want to rebuild doc automatically, we do not add it to the 'ALL' rule
+    message(AUTHOR_WARNING "The target `docs` was added, but not the default build rule. It will be triggered by `make docs` or `make package`")
+    add_custom_target(docs)
+  else()
+    # Building standalone, so add to ALL
+    message(AUTHOR_WARNING "The target `docs` was added to the default build rule")
+    add_custom_target(docs ALL)
+  endif()
```

Comment:
Handle whether it's built standalone or as part of E+. If E+, handle the `BUILD_DOCS_ONLY_WITH_PACKAGE`